src/bson.cc

```diff
@@ -337,11 +337,21 @@ void BSONSerializer<T>::SerializeValue(void *typeLocation,
   // Process all the values
   if (value->IsNumber()) {
     double doubleValue = NanTo<double>(value);
-    int intValue = (int)doubleValue;
-    if (intValue == doubleValue) {
+    double flooredValue = floor(doubleValue);
```

Comment:
teeny tiny suggestion, feel free to ignore, just makes this a bit more consistent with the line below it

```suggestion
    double flooredValue = std::floor(doubleValue);
```

Reply:
Worth doing, fixed 🙂